src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2007-2018, GoodData(R) Corporation. All rights reserved.
+ */
+package com.gooddata.executeafm.afm;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import static com.gooddata.util.Validate.notNull;
+
+/**
+ * The interface of the {@link MeasureDefinition} classes that are derived from the master measure and have the identifier of the master measure.
+ */
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = PopMeasureDefinition.class, name = PopMeasureDefinition.NAME),
+        @JsonSubTypes.Type(value = OverPeriodMeasureDefinition.class, name = OverPeriodMeasureDefinition.NAME),
+        @JsonSubTypes.Type(value = PreviousPeriodMeasureDefinition.class, name = PreviousPeriodMeasureDefinition.NAME)
+})
+public abstract class DerivedMeasureDefinition implements MeasureDefinition, Serializable {
+
+    private static final long serialVersionUID = -1203802872091017113L;
+
+    private final String measureIdentifier;
```

Comment:
I wonder, whether it should be `protected`?

Reply:
I made it protected and fixed its occurrences in the subclasses from getter to direct member access.